File: c++/GameScr.cpp
#include "GameScr.h"
#include <string>

using namespace std;

GameScr::GameScr()
{
	timeDisplayText = Utils::getTimeNow();
}

int marginTop = 5, widthButton = 80, heightText = 20;
//toa do x, y, do cao - do rong, khoang cach giua cac image them dong vat moi
int xImageAddAnimal, yImageAddAnimal1, heightImageAddAnimal1, distanceBetweenAddAnimal;

int GameScr::onInit() {
	dem = 0;
	gameState = isFree;
	level = 2;
    yImageAddAnimal1 = SCREEN_HEIGHT - 250;
    xImageAddAnimal = SCREEN_WIDTH - 80;
    distanceBetweenAddAnimal = 80;
	heightImageAddAnimal1 = 70;
	for (int i = 1; i <= TOTAL_TYPE_ANIMAL; i++) {
		string strTmp = imgAddAnimal + to_string(i) + ".png";
		char *str =(char *)strTmp.c_str();
    	textureAddAnimal[i-1].Init(str, heightImageAddAnimal1, heightImageAddAnimal1);
	}
	if(!textureBg.Init(imgBg, 0, 0))
		return FALSE;
	if(!texturePause.Init(imgPause, widthButton, widthButton))
		return FALSE;
	if(!texturePlay.Init(imgPlay, widthButton, widthButton))
		return FALSE;
	string strTmp = imgItemDrop + to_string(ID_EGG_PNG) + ".png";
	char *str =(char *)strTmp.c_str();
	if(!textureEgg.Init(str, 25, 27))
		return FALSE;

	strTmp = imgItemDrop + to_string(ID_SHIT_PNG) + ".png";
	char *str1 =(char *)strTmp.c_str();
	if(!textureShit.Init(str1, 30, 30))
		return FALSE;

	scoreEat = 0;
	scoreDropEggs = 0;
	scoreEatShit = 0;
	listAnimal.clear();
	listItemDrop.clear();
	basket = Object();
	basket.setTexture(imgBasket, 120, 100);
	basket.setXY((SCREEN_WIDTH - basket.getWidth()) >> 1, SCREEN_HEIGHT - basket.getHeight() - 40);
	return TRUE;
}

void GameScr::onPaint() {
	// paint score
	textureBg.Blit(0,0);
	if(gameState == isPause || gameState == isGameOver || gameState == isFree) {
		texturePlay.Blit(SCREEN_WIDTH - widthButton, 0);
		long timeNow = Utils::getTimeNow();
		if(timeNow - timeDisplayText <= 500) {
			if(gameState == isPause) {
				DrawString("CONTINUE", 450, marginTop + 3, 150, heightText, D3DCOLOR_ARGB(255,255,0,0));
			} else if(gameState == isGameOver) {
				DrawString("PLAY AGAIN", 430, marginTop + 3, 150, heightText, D3DCOLOR_ARGB(255,255,0,0));
			}else if(gameState == isFree) {
				DrawString("Click to start game", 360, marginTop, 150, heightText, D3DCOLOR_ARGB(255,255,0,0));
			}
		} else if(timeNow - timeDisplayText >= 1000) {
			timeDisplayText = Utils::getTimeNow();
		}
		if(dem >= 60) dem = 0;
	} else if(gameState == isPlaying) {
		texturePause.Blit(SCREEN_WIDTH - widthButton, 0);
	}
	DrawString("Score:", 5, marginTop, 70, heightText, D3DCOLOR_ARGB(255,0,0,0));
	string strTmp = to_string(scoreEat) + "";
	char* sScore = (char *)strTmp.c_str();
	DrawString(sScore, 73, marginTop, 50, heightText, D3DCOLOR_ARGB(255,50,2,255));
	// paint score drop egg
	textureEgg.Blit(130, marginTop);
	strTmp = to_string(scoreDropEggs) + "";
	char* sScore1 = (char *)strTmp.c_str();
	DrawString(sScore1, 160, marginTop, 30, heightText, D3DCOLOR_ARGB(255,255,2,62));
	// paint score eat shit
	textureShit.Blit(200, marginTop);
	strTmp = to_string(scoreEatShit) + "";
	char* sScore2 = (char *)strTmp.c_str();
	DrawString(sScore2, 230, marginTop, 30, heightText, D3DCOLOR_ARGB(255,255,2,62));
	DrawString("Level", 255, marginTop, 70, heightText, D3DCOLOR_ARGB(255,0,0,0));
	strTmp = to_string(level-1) + "";
	char* sScore3 = (char *)strTmp.c_str();
	DrawString(sScore3, 320, marginTop, 50, heightText, D3DCOLOR_ARGB(255,50,2,255));

	list<Animal>::iterator iter_name; // Khai báo con trỏ để duyệt
	for (iter_name = listAnimal.begin(); iter_name != listAnimal.end(); iter_name++) {
		(*iter_name).onPaint();
	}

	list<ItemDrop>::iterator iter_name2; // Khai báo con trỏ để duyệt
	for (iter_name2 = listItemDrop.begin(); iter_name2 != listItemDrop.end(); iter_name2++) {
		(*iter_name2).onPaint();
	}
	for (int i = 0; i < TOTAL_TYPE_ANIMAL; i++) {
		textureAddAnimal[i].Blit(xImageAddAnimal, yImageAddAnimal1 - i * distanceBetweenAddAnimal);
	}

	basket.onPaint();
}

void GameScr::onUpdate() {
	if(gameState == isPlaying) {
		if(KEY_LEFT_DOWN) {
				int xOlder = basket.getX();
				if(xOlder > 0) {
					basket.setX(xOlder - 10);
				}
		} else if(KEY_RIGHT_DOWN) {
				int xOlder = basket.getX();
				if(xOlder < (SCREEN_WIDTH - basket.getWidth())) {
					basket.setX(xOlder + 10);
				}
		} 
		basket.onUpdate();

		list<Animal>::iterator iter_name; // Khai báo con trỏ để duyệt
		for (iter_name = listAnimal.begin(); iter_name != listAnimal.end(); iter_name++) {
			(*iter_name).onUpdate();
			if((*iter_name).onPorm()) {
				int type = rand() % 2;
				ItemDrop itemDrop = ItemDrop(type + 1, 
					(*iter_name).getX() +(*iter_name).getWidth() / 2, 
					(*iter_name).getY() + (*iter_name).getHeight() / 2, level);

				listItemDrop.push_back(itemDrop);
			}
		}

		list<ItemDrop>::iterator iter_name2; // Khai báo con trỏ để duyệt
		for (iter_name2 = listItemDrop.begin(); iter_name2 != listItemDrop.end(); iter_name2++) {
			(*iter_name2).onUpdate();
		}
		
		onCheckEat();
	}
}

bool GameScr::onCheckGameOver() {

	if(scoreDropEggs >= TOTAL_MAX_DROP_EDD || scoreEatShit >= TOTAL_MAX_EAT_SHIT) {
		onGameOver();
		return true;
	}
	return false;
}

void GameScr::onCheckEat() {
	list<ItemDrop>::iterator iter_name2;
	// xóa những vật chạy quá màn hình
	for (iter_name2 = listItemDrop.begin(); iter_name2 != listItemDrop.end(); iter_name2++) {
		if((*iter_name2).getY() >= SCREEN_HEIGHT) {
			(*iter_name2).onDestroy();
			if((*iter_name2).getTypeItem() == ID_EGG_PNG) {
				scoreDropEggs ++;
			}
			if(iter_name2 != --listItemDrop.end()) {
				listItemDrop.erase(iter_name2++);
			} else {
				listItemDrop.erase(iter_name2);
				break;
			}
		}
	}

	for (iter_name2 = listItemDrop.begin(); iter_name2 != listItemDrop.end(); iter_name2++) {
		int x0 = (*iter_name2).getX();
		int y0 = (*iter_name2).getY();
		int w0 = (*iter_name2).getWidth();
		int h0 = (*iter_name2).getHeight();
		int x1 = basket.getX();
		int y1 = basket.getY();
		int w1 = basket.getWidth();
		int h1 = basket.getHeight();
		if(x0 + w0> x1 + 20 && x0 + 40 < x1 + w1 && y0 + h0 > y1 + 40 && y0 + 60 < y1 + h1) {
			// la trứng thì cộng điểm
			if((*iter_name2).getTypeItem() == ID_EGG_PNG) {
				scoreEat ++;
				if(scoreEat % NUM_EGG_EAT_NEXT_LEVEL == 0) {
					level++;
				}
			} else {
				scoreEatShit ++;
			}
			(*iter_name2).onDestroy();
			if(iter_name2 != --listItemDrop.end()) {
				listItemDrop.erase(iter_name2++);
			} else {
				listItemDrop.erase(iter_name2);
				break;
			}
		}
	}
}

void GameScr::WindowProcedure (HWND hWnd, UINT uMessage, WPARAM wParam, LPARAM lParam) {
	POINTS coords; //Mouse coords
	int xOlder = 0;
	//Check message
	switch(uMessage)
	{
	case WM_KEYUP:
		resetKeyboard();
		break;
	case WM_KEYDOWN:
		{
			switch (wParam) {
			case VK_LEFT:
				KEY_LEFT_DOWN = true;
				// cout << basket.getX();
				break;
			case VK_RIGHT:
				KEY_RIGHT_DOWN = true;
				// cout << basket.getX();
				break;
			case (int)'g':
			case (int)'G':
				addNewAnimal(ID_GA_PNG);
				break;
			case (int)'v':
			case (int)'V':
				addNewAnimal(ID_VIT_PNG);
				break;
			case (int)'n':
			case (int)'N':
				addNewAnimal(ID_NGAN_PNG);
				break;
			case VK_DOWN:
				break;
			case VK_SPACE:
				doChangeStateGame();
				break;
			}
			break;
		}
	case WM_MOUSEMOVE:
		//Get mouse coords
		coords = MAKEPOINTS (lParam);
		if(gameState == isPlaying) {
			if(coords.x < (SCREEN_WIDTH - basket.getWidth())) {
				basket.setX(coords.x);
			} else {
				basket.setX(SCREEN_WIDTH - basket.getWidth());
			}
		}
		break;

	case WM_LBUTTONDOWN:
		//Get mouse coords
		coords = MAKEPOINTS (lParam);
		if(gameState == isGameOver || gameState == isFree) {
			onStart();
		}
		break;	

	case WM_LBUTTONUP:
		//Get mouse coords
		coords = MAKEPOINTS (lParam);
		if(coords.x > (SCREEN_WIDTH - widthButton) && coords.x < SCREEN_WIDTH &&
			coords.y > 0 && coords.y < widthButton) {
				doChangeStateGame();
		} else {
			if(gameState == isPlaying) {
				for (int i = 0; i < TOTAL_TYPE_ANIMAL; i++) {
					/*int y1 = yImageAddAnimal1 - (i * distanceBetweenAddAnimal);
					int y2 = y1 + heightImageAddAnimal1;*/
    				if(coords.x > xImageAddAnimal && coords.x < SCREEN_WIDTH 
    						&& coords.y > (yImageAddAnimal1 - (i * distanceBetweenAddAnimal)) 
    						&&  coords.y <  (yImageAddAnimal1 - i * distanceBetweenAddAnimal + heightImageAddAnimal1)) {
    					addNewAnimal(i + 1);
    				}
    			}
			}
		}
		break;

	case WM_RBUTTONDOWN:
		//Get mouse coords
		coords = MAKEPOINTS (lParam);
		break;

	case WM_RBUTTONUP:
		//Get mouse coords
		coords = MAKEPOINTS (lParam);
		break;	
	}
}

void GameScr::doChangeStateGame() {
	dem = 0;
	if(gameState == isPause) {
		onResume();
	} else if(gameState == isPlaying) {
		onPause();
	} else if(gameState == isGameOver || gameState == isFree) {
		onStart();
	}
}

void GameScr::addNewAnimal(int type) {
	Animal animal = Animal(type, rand() % SCREEN_WIDTH - 80, rand() % 100 + 30);
	listAnimal.push_back(animal);
}

void GameScr::onResetData() {
	dem = 0;
	gameState = isFree;
	level = 2;
	scoreEat = 0;
	scoreDropEggs = 0;
	scoreEatShit = 0;
	listAnimal.clear();
	listItemDrop.clear();
}

void GameScr::onStart() {
	onResetData();
	addNewAnimal(rand()%TOTAL_TYPE_ANIMAL + 1);
	addNewAnimal(rand()%TOTAL_TYPE_ANIMAL + 1);
	addNewAnimal(rand()%TOTAL_TYPE_ANIMAL + 1);
	addNewAnimal(rand()%TOTAL_TYPE_ANIMAL + 1);
	super::onStart();
}

void GameScr::onResume() {
	super::onResume();
}

void GameScr::onPause() {
	super::onPause();
}

void GameScr::onGameOver() {
	if(gameState != isGameOver) {
		string strTmp = "Your score: " + to_string(scoreEat) + "\nPress the space key or on click the game screen to play again";
		char * str = (char *)strTmp.c_str();
		MessageBox(NULL, str, "Game over", MB_OK);
		super::onGameOver();
	}
}

void GameScr::onDestroy() {
	//basket.texture.texture = NULL;
}

File: android/src/game/view/GameScr.java
package game.view;

import game.font.BitmapFont;
import game.framework.Game;
import game.framework.GameScreen;
import game.framework.Graphics;
import game.framework.Image;
import game.framework.Input.KeyEvent;
import game.framework.Input.TouchEvent;
import game.framework.impl.AndroidPixmap;
import game.object.Object;
import game.object.Animal;
import game.object.ItemDrop;
import game.util.Constrain;
import game.util.ImageResource;
import game.util.Utils;

import java.util.ArrayList;
import java.util.List;

import android.graphics.Color;

public class GameScr extends GameScreen {

	Object basket;
	List<Animal> listAnimal = new ArrayList<Animal>();
	List<ItemDrop> listItemDrop  = new ArrayList<ItemDrop>();
	Image textureEgg, textureShit;
	Image texturePause, texturePlay, textureBg;
	Image textureAddAnimal[] = new AndroidPixmap[Constrain.TOTAL_TYPE_ANIMAL];
	int level;
	int scoreEat;
	int scoreDropEggs, scoreEatShit;
	long timeDisplayText;

	int marginTop = 5, widthButton = 50, heightText = 22, marginTopAlterText = 5, marginRightAlterText = 100;
	int speedDrop = 2;
	int yImageAddAnimal1, xImageAddAnimal, distanceBetweenAddAnimal;

    public GameScr(Game game) {
		super(game);
		timeDisplayText = System.currentTimeMillis();
		setGameState(GAME_STATE.isFree);
		if(SCREEN_WIDTH <= 480) {
			heightText = 22;
			marginTopAlterText = 30;
		}
		marginRightAlterText = SCREEN_WIDTH - widthButton - 10;
    }
    
    public void onInit() {
    	setGameState(GAME_STATE.isFree);
    	level = 1;
    	speedDrop = 5;
    	yImageAddAnimal1 = SCREEN_HEIGHT - 250;
    	xImageAddAnimal = SCREEN_WIDTH - 80;
    	distanceBetweenAddAnimal = 80;
    	for (int i = 1; i <= textureAddAnimal.length; i++) {
    		textureAddAnimal[i-1] = ImageResource.getImage(Constrain.imgAddAnimal + i + ".png", 70, 70);
		}
    	textureBg = ImageResource.getImage(Constrain.imgBg, SCREEN_WIDTH, SCREEN_HEIGHT);
    	texturePause = ImageResource.getImage(Constrain.imgPause, widthButton, widthButton);
    	texturePlay = ImageResource.getImage(Constrain.imgPlay, widthButton, widthButton);
    	String strTmp = Constrain.imgItemDrop + (Constrain.ID_EGG_PNG) + ".png";
    	textureEgg = ImageResource.getImage(strTmp, 30, 33);
    	strTmp = Constrain.imgItemDrop + (Constrain.ID_SHIT_PNG) + ".png";
    	textureShit = ImageResource.getImage(strTmp, 30, 33);
    	scoreEat = 0;
    	scoreDropEggs = 0;
    	scoreEatShit = 0;
    	listAnimal.clear();
    	listItemDrop.clear();
    	basket = new Object();
    	basket.setImage(ImageResource.getImage(Constrain.imgBasket));
    	basket.setWidth(120);
    	basket.setHeight(100);
    	basket.setXY((SCREEN_WIDTH - basket.getWidth()) >> 1, SCREEN_HEIGHT - basket.getHeight() - 60);
    	basket.onMoveToX(basket.getX());
    }
	    
    public void paint(Graphics g) {
    	g.drawImage(textureBg, 0, 0, 0);

    	for (Animal iter_name : listAnimal) {
    		iter_name.onPaint(g);
		}

    	for (ItemDrop iter_name : listItemDrop) {
    		iter_name.onPaint(g);
		}
    	
    	if(isPause() || isGameOver() || isFree()) {
    		g.drawImage(texturePlay, SCREEN_WIDTH - widthButton, 0, 0);
    		long timeNow = System.currentTimeMillis();
    		if(timeNow - timeDisplayText <= 500) {
    			if(isPause()) {
    				BitmapFont.drawBoldFont(g, "CONTINUE", marginRightAlterText, marginTopAlterText, heightText, Color.argb(255,255,0,0), Graphics.RIGHT);
    			} else if(isGameOver()) {
    				BitmapFont.drawBoldFont(g, "PLAY AGAIN", marginRightAlterText, marginTopAlterText, heightText, Color.argb(255,255,0,0), Graphics.RIGHT);
    			}else if(isFree()) {
    				BitmapFont.drawBoldFont(g, "Touch on screen to start game", SCREEN_WIDTH >> 1, SCREEN_HEIGHT >> 1, heightText, Color.argb(255,255,0,0), Graphics.VCENTER);
    			}
    		} else if(timeNow - timeDisplayText >= 1000) {
    			timeDisplayText = System.currentTimeMillis();
    		}
    	} else if(isPlaying()) {
    		g.drawImage(texturePause, SCREEN_WIDTH - widthButton, 0, 0);
    	}
    	BitmapFont.drawBoldFont(g, "Score:", 5, marginTop, heightText, Color.argb(255,0,0,0), 0);
//    	String strTmp = scoreEat + "";
    	BitmapFont.drawBoldFont(g, scoreEat + "", 80, marginTop, heightText, Color.argb(255,50,2,255), 0);
    	// paint score drop egg
    	g.drawImage(textureEgg, 130, marginTop, 0);
    	BitmapFont.drawBoldFont(g, scoreDropEggs + "", 163, marginTop, heightText, Color.argb(255,255,2,62), 0);
    	// paint score eat shit
    	g.drawImage(textureShit, 200, marginTop, 0);
    	BitmapFont.drawBoldFont(g, scoreEatShit + "", 230, marginTop, heightText, Color.argb(255,255,2,62), 0);
    	BitmapFont.drawBoldFont(g, "Level", 255, marginTop, heightText, Color.argb(255,0,0,0), 0);
    	BitmapFont.drawBoldFont(g, (level) + "", 320, marginTop, heightText, Color.argb(255,50,2,255), 0);
    	
    	for (int i = 0; i < textureAddAnimal.length; i++) {
			g.drawImage(textureAddAnimal[i], xImageAddAnimal, yImageAddAnimal1 - i * distanceBetweenAddAnimal,
					Graphics.BOTTOM);
		}
    	
    	basket.onPaint(g);
    	
		if(isGameOver()) {
			g.setColor(0xff000000);
			g.fillRoundRect((SCREEN_WIDTH - 450) >> 1, (SCREEN_HEIGHT - 200) >> 1, 450, 200, 15, 15, 130);
			BitmapFont.drawBoldFont(g, "GAME OVER", SCREEN_WIDTH >> 1, (SCREEN_HEIGHT - (heightText * 4)) >> 1, heightText + 4, Color.argb(255,255,0,0), Graphics.VCENTER);
			BitmapFont.drawBoldFont(g, "Your score: " + scoreEat, SCREEN_WIDTH >> 1, (SCREEN_HEIGHT - (heightText)) >> 1, heightText + 4, "#ffffff", Graphics.VCENTER);
//			BitmapFont.drawBoldFont(g, "Touch on screen to play again", SCREEN_WIDTH >> 1, (SCREEN_HEIGHT + (heightText << 1)) >> 1, heightText, "#ffffff", Graphics.VCENTER);
			
		}

    }


    public void update() {
		try {
			List<TouchEvent> touchEvents = game.getInput().getTouchEvents();
			int len = touchEvents.size();
			for (int i = 0; i < len; i++) {
				TouchEvent event = touchEvents.get(i);
				if (event.type == TouchEvent.TOUCH_DOWN) {
					pointerPressed(event.x, event.y);
				}
				if (event.type == TouchEvent.TOUCH_DRAGGED) {
					pointerDragged(event.x, event.y);
				}
				if (event.type == TouchEvent.TOUCH_UP) {
					pointerReleased(event.x, event.y);
				}
			}

			List<KeyEvent> keyEvents = game.getInput().getKeyEvents();

			for (int i = 0; i < keyEvents.size(); i++) {
				KeyEvent keyEvent = keyEvents.get(i);
				if (keyEvent.type == keyEvent.KEY_DOWN) {

				}
				if (keyEvent.type == keyEvent.KEY_UP) {

				}
			}
		} catch (Exception e) {
			e.printStackTrace();
		}
		if(isPlaying()) {
			basket.onUpdate();
			for (Animal iter_name : listAnimal) {
				(iter_name).onUpdate();
				if((iter_name).onPorm()) {
					int type = Utils.getRandom(0, 10) % 2;
					ItemDrop itemDrop = new ItemDrop(type + 1, 
						(iter_name).getX() +(iter_name).getWidth() / 2, 
						(iter_name).getY() + (iter_name).getHeight() / 2, speedDrop);

					listItemDrop.add(itemDrop);
				}
			}
			for (ItemDrop iter_name : listItemDrop) {
				(iter_name).onUpdate();
			}
			
			onCheckEat();
		}
	}

	boolean onCheckGameOver() {
	
		if(scoreDropEggs >= Constrain.TOTAL_MAX_DROP_EDD || scoreEatShit >= Constrain.TOTAL_MAX_EAT_SHIT) {
			gameOver();
			return true;
		}
		return false;
	}
	
	void onCheckEat() {
		for (int i = 0; i < listItemDrop.size(); i++) {
			ItemDrop iter_name2 = listItemDrop.get(i);
			if((iter_name2).getY() >= SCREEN_HEIGHT) {
				(iter_name2).onDestroy();
				if((iter_name2).getTypeItem() == Constrain.ID_EGG_PNG) {
					scoreDropEggs ++;
					if(onCheckGameOver()) {
						gameOver();
					}
				}
				listItemDrop.remove(i);
				i--;
			} else {
				int x0 = (iter_name2).getX();
				int y0 = (iter_name2).getY();
				int w0 = (iter_name2).getWidth();
				int h0 = (iter_name2).getHeight();
				int x1 = basket.getX();
				int y1 = basket.getY();
				int w1 = basket.getWidth();
				int h1 = basket.getHeight();
				if(x0 + w0> x1 + 20 && x0 + 30 < x1 + w1 && y0 + h0 > y1 + 40 && y0 + 60 < y1 + h1) {
					// la trứng thì cộng điểm
					if((iter_name2).getTypeItem() == Constrain.ID_EGG_PNG) {
						scoreEat ++;
						if(scoreEat % Constrain.NUM_EGG_EAT_NEXT_LEVEL == 0) {
							speedDrop += 3;
							level ++;
						}
					} else {
						scoreEatShit ++;
						if(onCheckGameOver()) {
							gameOver();
						}
					}
					listItemDrop.remove(i);
					i--;
				}
			}
		}
	}

	void doChangeStateGame() {
		if(isPause()) {
			resume();
		} else if(isPlaying()) {
			pause();
		} else if(isGameOver() || isFree()) {
			start();
		}
	}
	
	public void addNewAnimal(int type) {
		Animal animal = new Animal(type, Utils.getRandom(0, GameScreen.SCREEN_WIDTH - 100), Utils.getRandom(50, 150));
		listAnimal.add(animal);
	}
	
	void onResetData() {
		setGameState(GAME_STATE.isFree);
		level = 1;
		speedDrop = 5;
		scoreEat = 0;
		scoreDropEggs = 0;
		scoreEatShit = 0;
		listAnimal.clear();
		listItemDrop.clear();
	}

	public void start() {
		onResetData();
		addNewAnimal(Utils.getRandom(0, 100)%Constrain.TOTAL_TYPE_ANIMAL + 1);
		addNewAnimal(Utils.getRandom(0, 100)%Constrain.TOTAL_TYPE_ANIMAL + 1);
		addNewAnimal(Utils.getRandom(0, 100)%Constrain.TOTAL_TYPE_ANIMAL + 1);
		addNewAnimal(Utils.getRandom(0, 100)%Constrain.TOTAL_TYPE_ANIMAL + 1);
//		addNewAnimal(Utils.getRandom(0, 100)%Constrain.TOTAL_TYPE_ANIMAL + 1);
//		addNewAnimal(Utils.getRandom(0, 100)%Constrain.TOTAL_TYPE_ANIMAL + 1);
//		addNewAnimal(Utils.getRandom(0, 100)%Constrain.TOTAL_TYPE_ANIMAL + 1);
//		addNewAnimal(Utils.getRandom(0, 100)%Constrain.TOTAL_TYPE_ANIMAL + 1);
//		addNewAnimal(Utils.getRandom(0, 100)%Constrain.TOTAL_TYPE_ANIMAL + 1);
//		addNewAnimal(Utils.getRandom(0, 100)%Constrain.TOTAL_TYPE_ANIMAL + 1);
//		addNewAnimal(Utils.getRandom(0, 100)%Constrain.TOTAL_TYPE_ANIMAL + 1);
//		addNewAnimal(Utils.getRandom(0, 100)%Constrain.TOTAL_TYPE_ANIMAL + 1);
//		addNewAnimal(Utils.getRandom(0, 100)%Constrain.TOTAL_TYPE_ANIMAL + 1);
//		addNewAnimal(Utils.getRandom(0, 100)%Constrain.TOTAL_TYPE_ANIMAL + 1);
//		addNewAnimal(Utils.getRandom(0, 100)%Constrain.TOTAL_TYPE_ANIMAL + 1);
//		addNewAnimal(Utils.getRandom(0, 100)%Constrain.TOTAL_TYPE_ANIMAL + 1);
//		addNewAnimal(Utils.getRandom(0, 100)%Constrain.TOTAL_TYPE_ANIMAL + 1);
//		addNewAnimal(Utils.getRandom(0, 100)%Constrain.TOTAL_TYPE_ANIMAL + 1);
//		addNewAnimal(Utils.getRandom(0, 100)%Constrain.TOTAL_TYPE_ANIMAL + 1);
//		addNewAnimal(Utils.getRandom(0, 100)%Constrain.TOTAL_TYPE_ANIMAL + 1);
//		addNewAnimal(Utils.getRandom(0, 100)%Constrain.TOTAL_TYPE_ANIMAL + 1);
//		addNewAnimal(Utils.getRandom(0, 100)%Constrain.TOTAL_TYPE_ANIMAL + 1);
//		addNewAnimal(Utils.getRandom(0, 100)%Constrain.TOTAL_TYPE_ANIMAL + 1);
//		addNewAnimal(Utils.getRandom(0, 100)%Constrain.TOTAL_TYPE_ANIMAL + 1);
		super.start();
	}


    protected void pointerDragged(int x, int y) {
    	if(y > 200) {
    		basket.onMoveToX(x);
    	}
    }

    protected void pointerPressed(int x, int y) {
    	if(isFree() || isGameOver()) {
    		start();
    	} else if(isPlaying()) {
    		if(x > SCREEN_WIDTH - widthButton - 5 && x < SCREEN_WIDTH && y < widthButton + 5) {
    			pause();
    		} else {
    			for (int i = 0; i < textureAddAnimal.length; i++) {
    				if(x > xImageAddAnimal && x < SCREEN_WIDTH 
    						&& y > (yImageAddAnimal1 - (i+1) * distanceBetweenAddAnimal) 
    						&&  y <  (yImageAddAnimal1 - i * distanceBetweenAddAnimal)) {
    					addNewAnimal(i + 1);
    				}
    			}
    		}
    	} else if(isPause()) {
    		if(x > SCREEN_WIDTH - widthButton - 5 && x < SCREEN_WIDTH && y < widthButton + 5) {
    			resume();
    		}
    	}
    }

    protected void pointerReleased(int x, int y) {

    }

	@Override
	public void gameOver() {
		super.gameOver();
	}

	@Override
	public void pause() {
		super.pause();
	}

	@Override
	public void resume() {
		super.resume();
	}

	@Override
	public void dispose() {
		
	}
}
